sklearn/decomposition/tests/test_pca.py

```diff
@@ -317,22 +323,22 @@ def test_n_components_mle_error(svd_solver):
         pca.fit(X)
 
 
-def test_pca_dim():
+def test_pca_dim(global_random_seed):
     # Check automated dimensionality setting
-    rng = np.random.RandomState(0)
-    n, p = 100, 5
+    rng = np.random.RandomState(global_random_seed)
+    n, p = 250, 5
```

Comment:
Only adding the `global_random_seed` fixture showed, that this test was seed-sensitive. Since the MLE cannot be expected to yield the "true" number of component for any dataset, this is not surprising.
Increasing the number of samples makes the test seed-insensitive in the 0-99 range.